xclim/ensembles/_robustness.py

```diff
@@ -0,0 +1,240 @@
+
+        x = np.insert(
+            x1, x2_in_1, x2
+        )  # Merge to get all "discontinuities" of the CDF difference
+        y1_f = np.insert(
+            y1, x2_in_1, np.r_[0, y1][x2_in_1]
+        )  # y1 with repeated value (to the right) where x2 is inserted
+        y2_f = np.insert(
+            y2, x1_in_2, np.r_[0, y2][x1_in_2]
+        )  # Same for y2. 0s are prepended where needed.
+
+        # Discrete integral of the squared difference (distance) between the two CDFs.
+        return np.sum(np.diff(x) * (y1_f - y2_f)[:-1] ** 2)
+
+    # Get sorted vectors
+    v_sims = np.sort(sims.flatten())  # "cumulative" models distribution
+    v_means = np.sort(sims.mean(axis=-1))  # Multi-model mean
+    v_hist = np.sort(hist)  # Historical values
+
+    A1 = diff_cdf_sq_area_int(v_sims, v_means)
+    A2 = diff_cdf_sq_area_int(v_hist, v_means)
+
+    return 1 - A1 / A2
+
+
+@metric(
+    {
+        0: "No significant change",
+        1: "No agreement on sign of change",
+        2: "Agreement on sign of change",
+    },
+    hist_dims=["realization"],
+)
+def tebaldi_et_al(hist, sims, X=0.5, Y=0.8, p_change=0.01):
+    """Robustness categories from Tebaldi et al. (2011).
+
+    Compare the historical mean to the projected values for each model.
+    Categories:
+      - 0 : less than X% of the models show significant change.
+      - 1 : models show significant change, but less then Y% agree on sign of change.
+      - 2 : models show significant change and agree on sign of change.
+
+    Parameters
+    ----------
+    hists : ndarray
+      Multi-model historical mean along 'realization' (nr).
+    sims : ndarray
+      Simulated values along 'realization' and 'time' (nr, nt).
+    X : float, 0 .. 1
+      Threshold fraction of models agreeing on significant change.
+    Y : float, 0 .. 1
+      Threshold fraction of models agreeing on the sign of the change.
+    p_change : float
+      p-value threshold for rejecting the hypothesis of no significant
+      change in the t-test.
+
+    Returns
+    -------
+    int
+        Categories of agreement on the change and its sign.
+
+    References
+    ----------
+    Tebaldi C., Arblaster, J.M. and Knutti, R. (2011) Mapping model agreement on future climate projections. GRL. doi:10.1029/2011GL049863
+    """
+    # Test hypothesis of no significant change
+    t, p = spstats.ttest_1samp(sims, hist, axis=1)
+
+    # When p < p_change, the hypothesis of no significant change is rejected.
+    # We need at least X % models showing significant change
+    if (p < p_change).sum() / hist.size <= X:
+        return 0  # No significant change
+
+    # Test that models agree on the sign of the change
+    change_sign = np.sign(sims.mean(axis=1) - hist).clip(0, 1)
+    if (1 - Y) <= change_sign.sum() / hist.size <= Y:
+        return 1  # Significant change but no agreement on sign of change
+    return 2  # Significant change and agreement on sign of change
```

Comment:
Thinking about it, `change_sign.sum() / hist.size` itself would be a very useful output for the function. This could allow the user to create their own thresholds of significance (for example: Y>0.8 = "significative", Y>0.95 = "very significative"), instead of the binary `1` and `2` results.